receiver/dockerstatsreceiver/testdata/mock/two_containers/expected_metrics.yaml

```diff
@@ -4,24 +4,24 @@ resourceMetrics:
         - key: container.runtime
           value:
             stringValue: docker
+        - key: container.hostname
```

Comment:
Looks like these two metrics have switched order in the expected, which made the diff a bit harder to reason with. Do you know why this happened? I felt like this had deterministic order before. 

Do you think you can make it the same order? Because if we merge this it will create diffs in other PRs that have affected the same output

Reply:
I just tried and regenerate all the mock files. We use this [line](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/6c71da24ffbf823e3cf4530d91e32b596f7c6b77/receiver/dockerstatsreceiver/receiver_test.go#L250) to create them, without any manual intervention. So, I don't know why the order has changed...

I see there were some changes (https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/22717, https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/22916) recently in the golden package regarding writing metrics, maybe it's because of this.